app/Importers/DataImporter.php

```diff
@@ -99,19 +110,26 @@ protected function createImportInfoDetails(string $code, string $providerName):
 
     protected function load(string $cityName, string $countryName, string $lat = "", string $long = ""): string
     {
-        $city = City::query()->where("name", $cityName)->first();
-        $alternativeCityName = CityAlternativeName::query()->where("name", $cityName)->first();
+        $country = Country::query()->where("name", $countryName)->orWhere("alternative_name", $countryName)->first();
 
-        if ($city || $alternativeCityName) {
-            $cityId = $city ? $city->id : $alternativeCityName->city_id;
+        if ($country) {
+            $city = City::query()->where("name", $cityName)->where("country_id", $country->id)->first();
+            $alternativeCityName = CityAlternativeName::query()->where("name", $cityName)->first();
 
-            $this->createProvider($cityId, self::getProviderName());
+            if ($city) {
+                $cityId = $city->id;
+                $this->createProvider($cityId, self::getProviderName());
```

Comment:
Why method `getProviderName()` is static?

Reply:
Changed it
I wanted to use `get_class($this)` but composer forces `static::class`

Reply:
Ok fine. This change will break all importers.

Reply:
Undone it

Reply:
Lets keep it as it is.